data/model/oauth.py

```diff
@@ -378,12 +399,42 @@
     return query
 
 
+def get_assigned_authorization_for_user(user_obj, uuid):
+    try:
+        assigned_token = (
+            OauthAssignedToken.select()
+            .join(OAuthApplication)
+            .where(OauthAssignedToken.assigned_user == user_obj, OauthAssignedToken.uuid == uuid)
```

Comment:
isn't the uuid unique? does it need to be combined with the user to get the assigned_token?

Reply:
It's used to express `token that exists for this user with this ID`. I could use just the uuid but then i would need an additional check in python for the `assigned_user` field. This basically moves the condition check to the DB. Same for the other lookup.